sov-modules/sov-modules-impl/prover-incentives/src/tests.rs

```diff
@@ -0,0 +1,195 @@
+use sovereign_sdk::core::mocks::{MockCodeCommitment, MockProof, MockZkvm};
+
+use crate::ProverIncentives;
+
+type C = DefaultContext;
+
+const BOND_AMOUNT: u64 = 1000;
+const MOCK_CODE_COMMITMENT: MockCodeCommitment = MockCodeCommitment([0u8; 32]);
+
+pub fn generate_address(key: &str) -> <C as Spec>::Address {
+    let hash = <C as Spec>::Hasher::hash(key.as_bytes());
+    Address::from(hash)
+}
+
+fn create_bank_config() -> (bank::BankConfig<C>, <C as Spec>::Address) {
+    let prover_address = generate_address("prover_pub_key");
+
+    let token_config = bank::TokenConfig {
+        token_name: "InitialToken".to_owned(),
+        address_and_balances: vec![(prover_address.clone(), BOND_AMOUNT * 5)],
+    };
+
+    (
+        bank::BankConfig {
+            tokens: vec![token_config],
+        },
+        prover_address,
+    )
+}
+
+fn setup(
+    working_set: &mut WorkingSet<<C as Spec>::Storage>,
+) -> (ProverIncentives<C, MockZkvm>, Address) {
+    // Initialize bank
+    let (bank_config, prover_address) = create_bank_config();
+    let bank = bank::Bank::<C>::new();
+    bank.genesis(&bank_config, working_set)
+        .expect("bank genesis must succeed");
+
+    let token_address = bank::create_token_address::<C>(
+        &bank_config.tokens[0].token_name,
+        &bank::genesis::DEPLOYER,
+        bank::genesis::SALT,
+    );
+
+    // initialize prover incentives
+    let module = ProverIncentives::<C, MockZkvm>::new();
+    let config = crate::ProverIncentivesConfig {
+        bonding_token_address: token_address,
+        minimum_bond: BOND_AMOUNT,
+        commitment_of_allowed_verifier_method: MockCodeCommitment([0u8; 32]),
+        initial_provers: vec![(prover_address.clone(), BOND_AMOUNT)],
+    };
+
+    module
+        .genesis(&config, working_set)
+        .expect("prover incentives genesis must succeed");
+    (module, prover_address)
+}
+
+#[test]
+fn test_burn_on_invalid_proof() {
+    let storage = ProverStorage::temporary();
+    let mut working_set = WorkingSet::new(storage);
+    let (module, prover_address) = setup(&mut working_set);
+
+    // Assert that the prover has the correct bond amount before processing the proof
+    assert_eq!(
+        module
+            .get_bond_amount(prover_address.clone(), &mut working_set)
+            .value,
+        BOND_AMOUNT
+    );
+
+    // Process an invalid proof
+    {
+        let context = DefaultContext {
+            sender: prover_address.clone(),
+        };
+        let proof = MockProof {
+            program_id: MOCK_CODE_COMMITMENT,
+            is_valid: false,
+            log: &[],
+        };
+        module
+            .process_proof(proof.encode_to_vec().as_ref(), &context, &mut working_set)
+            .expect("An invalid proof is not an error");
+    }
+
+    // Assert that the prover's bond amount has been burned
+    assert_eq!(
+        module
+            .get_bond_amount(prover_address.clone(), &mut working_set)
+            .value,
+        0
```

Comment:
Is prover slashed 100% all the time?

Reply:
No, the prover is just slashed the minimum bond amount - but we've already asserted that they're bonded exactly that amount on line 70; so the new balance must be 0